utt/signature-processor/include/sigProcessor.hpp

```diff
@@ -0,0 +1,181 @@
+   * signature.
+   *
+   */
+ public:
+  /**
+   * @brief JobCoordinatorCb is a callback for choosing the coordinator for a given signature job. The only requirement
+   * is that it has to be deterministic among all replicas The default implementation is simply job_id % n
+   */
+  typedef std::function<uint16_t(uint64_t)> JobCoordinatorCb;
+
+  /**
+   * @brief GenerateAppClientRequestCb is a callback for generating the content of ClientRequestMsg that will be
+   * published by the replica who managed to collect the full signature. The output might be a platform specific (for
+   * example, a composed ethereum request) and has to contain the full signature and its id. The default implementation
+   * is simply the [sig_id, signature] serialized as a vector of bytes
+   *
+   */
+  typedef std::function<std::vector<uint8_t>(uint64_t, const libutt::api::types::Signature&)>
+      GenerateAppClientRequestCb;
+
+  /**
+   * @brief Construct a new Sig Processor object
+   *
+   * @param repId replica id, has to be identical to the bft replica ID
+   * @param n the number of replicas
+   * @param threshold threshold for the number of required partial signatures (typically f+1)
+   * @param timer_handler_timeout the time elapsed before the timer triggers.
+   * @param msgsCommunicator a MsgsCommunicator object (should be taken from the bft layer)
+   * @param msgHandlers a MsgHandlersRegistrator object (should be taken from the bft layer)
+   * @param timers a Timers object (should be taken from the bft layer)
+   */
+  SigProcessor(uint16_t repId,
+               uint16_t n,
+               uint16_t threshold,
+               uint64_t timer_handler_timeout,
+               std::shared_ptr<bftEngine::impl::MsgsCommunicator> msgsCommunicator,
+               std::shared_ptr<bftEngine::impl::MsgHandlersRegistrator> msgHandlers,
+               concordUtil::Timers& timers);
+
+  /**
+   * @brief Initiates the protocol for computing the full signature for a given RCM. This is a thread safe method and it
+   * can be called by any thread.
+   *
+   * @param sigId id for the given job
+   * @param pidHash the pid hash of the client
+   * @param rcm1 the partial rcm (generated by the client)
+   * @param s2 the replicas secret share (has to be non predictable and deterministic among all replicas)
+   * @param signer the UTT Registrator object
+   * @param job_timeout timeout for this job. If the timeout passes, the replica will broadcast the partial signature to
+   * all. default value is 1000 ms.
+   */
+  void processRCMSignature(uint64_t sigId,
+                           const libutt::api::types::CurvePoint& pidHash,
+                           const libutt::api::Commitment& rcm1,
+                           const libutt::api::types::CurvePoint& s2,
+                           std::shared_ptr<libutt::api::Registrator> signer,
+                           uint64_t job_timeout = 1000);
```

Comment:
job_timeout lacks time resolution, consider "job_timeout_ms" for example.

Reply:
fixed